### PR TITLE
Configure tslint

### PR DESCRIPTION
**Pull request by edgarmueller**
This is run as a pre-build step, i.e. it won't be executed when running the tests.

### CODE REVIEW

File: src/components/renderers/controls/control.html [hunk 1]
```diff
@@ -4,7 +4,7 @@
     </div>
     <div ng-transclude>
     </div>
-    <div>
+    <div popover="Hello foo" popover-is-open="vm.alerts">
```
Comment (eneufeld) [inline src/components/renderers/controls/control.html]: is this intended?

File: src/components/renderers/controls/enum/enum-directive.ts [hunk 2]
```diff
@@ -1,45 +1,58 @@
...
+                                                         dataSchema: any,
+                                                         dataObject: any,
+                                                         pathResolver: IPathResolver ){
+    if (element.type !== 'Control') {
         return NOT_FITTING;
-    let currentDataSchema=pathResolver.resolveSchema(dataSchema,element['scope']['$ref']);
-    if(currentDataSchema == undefined || !currentDataSchema.hasOwnProperty('enum'))
+    }
+    let currentDataSchema = pathResolver.resolveSchema(dataSchema, element['scope']['$ref']);
+    if (currentDataSchema === undefined || !currentDataSchema.hasOwnProperty('enum')) {
```
Comment (eneufeld) [inline src/components/renderers/controls/enum/enum-directive.ts]: why not use lodash _has

File: src/components/renderers/controls/reference/reference-directive.ts [hunk 3]
```diff
@@ -1,44 +1,46 @@
...
 class ReferenceController extends AbstractControl {
-    static $inject = ['$scope','PathResolver'];
-    constructor(scope:ReferenceControllerScopepe,refResolver: IPathResolver) {
-        super(scope,refResolver);
+    static $inject = ['$scope', 'PathResolver'];
+    constructor(scope: ReferenceControllerScope, pathResolver: IPathResolver) {
+        super(scope, pathResolver);
     }
-    private get link(){
+    public get link(){
```
Comment (eneufeld) [inline src/components/renderers/controls/reference/reference-directive.ts]: why public?

File: src/components/renderers/layouts/masterdetail/masterdetail-directives.ts [hunk 4]
```diff
@@ -1,125 +1,159 @@
...
+    private selectedChild: any;
+    private selectedSchema: SchemaElement;
+    constructor(scope: MasterDetailControllerScope, pathResolver: IPathResolver) {
+        super(scope, pathResolver);
         this.subSchema = this.pathResolver.resolveSchema(this.schema, this.schemaPath);
     }
-    private select(selectedChild:any,selectedSchema:SchemaElement){
-        this.selectedChild=selectedChild;
-        this.selectedSchema=selectedSchema;
+    public select(selectedChild: any, selectedSchema: SchemaElement) {
```
Comment (eneufeld) [inline src/components/renderers/layouts/masterdetail/masterdetail-directives.ts]: why public

File: src/components/renderers/layouts/masterdetail/masterdetail-directives.ts [hunk 5]
```diff
@@ -1,125 +1,159 @@
...
+                                         ng-repeat="child in instance[key]" 
+                                         ng-class="{'
+                                           jsf-masterdetail-empty':!hasKeys(value.items)
+                                         }">
                             <accordion-heading>
-                                <span ng-click="selectElement(child,value)" ng-class="{'jsf-masterdetail-selected':selectedChild==child}">{{child.name!=undefined?child.name:child}}<!-- label provider needed --></span>
-                                <i class="pull-right glyphicon" ng-class="{'glyphicon-chevron-down': status_object.open, 'glyphicon-chevron-right': !status_object.open}" ng-if="hasKeys(value.items)"></i>
+                                <span ng-click="selectElement(child,value)" 
+                                      ng-class="{
+                                         'jsf-masterdetail-selected':selectedChild==child
```
Comment (eneufeld) [inline src/components/renderers/layouts/masterdetail/masterdetail-directives.ts]: strange formatting ...

File: src/components/renderers/layouts/masterdetail/masterdetail-directives.ts [hunk 6]
```diff
@@ -1,125 +1,159 @@
...
+        select: '&',
+        filter: '&'
     };
     link = (scope, element) => {
-        if(Object.keys(scope.filter(scope.childSchema.properties)).length!=0){
-            this.$compile('<jsonforms-masterdetail-collection select="select" properties="childSchema.properties" instance="childData"></jsonforms-masterdetail-collection>')
-            (scope, (cloned) => {
-                element.replaceWith(cloned);
-            });
+        if (Object.keys(scope.filter(scope.childSchema.properties)).length !== 0) {
```
Comment (eneufeld) [inline src/components/renderers/layouts/masterdetail/masterdetail-directives.ts]: lodash _keys?

File: src/components/renderers/renderer-service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+import IRootScopeService = angular.IRootScopeService;
+import ICompileService = angular.ICompileService;
+
+describe("Renderer service", () => {
+
+
+    beforeEach(angular.mock.module('jsonforms.form'));
+
+    xit("should support remote references",
```
Comment (eneufeld) [inline src/components/renderers/renderer-service.spec.ts]: xit?

**Comment by eneufeld** [comment]
looks good, thanks.
Mainly a question about the control.html change

File: src/components/renderers/controls/reference/reference-directive.ts (hunk 3 above)
Comment (edgarmueller) [inline src/components/renderers/controls/reference/reference-directive.ts]: In order to remove compiler warnings about unused methods.

File: src/components/renderers/controls/control.html (hunk 1 above)
Comment (edgarmueller) [inline src/components/renderers/controls/control.html]: Nope, this is a remain-over. Thanks for pointing it out.

File: src/components/renderers/layouts/masterdetail/masterdetail-directives.ts (hunk 5 above)
Comment (edgarmueller) [inline src/components/renderers/layouts/masterdetail/masterdetail-directives.ts]: the current max line length is set to 100, so I had to come up with something ;)

File: src/components/renderers/renderer-service.spec.ts (hunk 7 above)
Comment (edgarmueller) [inline src/components/renderers/renderer-service.spec.ts]: This test will fail, I should've probably pushed it in a separate commit. `xit` causes the test to be ignored.

**Pull request merged**